packages/typespec-ts/test/modularUnit/scenarios/samples/parameters/bodyOptionalParameterName.md

```diff
@@ -0,0 +1,258 @@
+# Should generate optional body in option parameter
+
+Should generate optional body in option parameter.
+
+## TypeSpec
+
+This is tsp definition.
+
+```tsp
+
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+using Azure.ClientGenerator.Core;
+
+@armProviderNamespace
+@service(#{ title: "HardwareSecurityModules" })
+@versioned(Versions)
+@armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v6)
+namespace Microsoft.HardwareSecurityModules;
+
+enum Versions {
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  v2021_10_01_preview: "2021-10-01-preview",
+}
+
+model CloudHsmClusterProperties {
+  statusMessage?: string;
+}
+
+model CloudHsmCluster
+  is Azure.ResourceManager.TrackedResource<CloudHsmClusterProperties> {
+  ...ResourceNameParameter<
+    Resource = CloudHsmCluster,
+    KeyName = "cloudHsmClusterName",
+    SegmentName = "cloudHsmClusters",
+    NamePattern = "^[a-zA-Z0-9-]{3,23}$"
+  >;
+
+  identity?: Azure.ResourceManager.CommonTypes.ManagedServiceIdentity;
+}
+
+model BackupRequestProperties extends BackupRestoreRequestBaseProperties {}
+
+model BackupRestoreRequestBaseProperties {
+
+  azureStorageBlobContainerUri: url;
+
+  @secret
+  token?: string;
+}
+
+model BackupResult {
+  properties?: BackupResultProperties;
+}
+
+model BackupResultProperties {
+
+  azureStorageBlobContainerUri?: url;
+
+  backupId?: string;
+}
+
+@armResourceOperations
+interface CloudHsmClusters {
+  backup is ArmResourceActionAsync<
+    CloudHsmCluster,
+    BackupRequestProperties,
+    ArmResponse<BackupResult> & Azure.Core.RequestIdResponseHeader,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = BackupResult> &
+      ArmLroLocationHeader &
+      Azure.Core.Foundations.RetryAfterHeader &
+      Azure.Core.RequestIdResponseHeader,
+    OptionalRequestBody = true
+  >;
+}
+@@clientName(CloudHsmClusters.backup::parameters.body,
```

Comment:
Could you add test cases for other optional paramter e.g path/query/header?

Reply:
query exists in https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularUnit/scenarios/samples/parameters/paramaterSpreadCheck.md